File: py/deckRevealedIncreasing.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        def shiftandInsert(d:list[int], newdeck):
            toshift = newdeck[-1]
            newdeck.pop()
            num = d[-1]
            d.pop()
            newdeck.insert(0, toshift)
            newdeck.insert(0, num)
            

        deck.sort()
        newdeck = [deck[-1]]
        deck.pop()
        for i in reversed(range(-1, len(deck) - 1)):
            shiftandInsert(deck, newdeck)
        
        return newdeck

File: cpp/deckRevealedIncreasing.cpp
#include <vector>
#include <deque>
#include <algorithm>

using namespace std;
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        deque<int> dq;
        sort(deck.begin(), deck.end());

        for (auto it = deck.rbegin(); it != deck.rend(); ++it) {
            if (!dq.empty()) {
                int last = dq.back();
                dq.pop_back();
                dq.push_front(last);
            }
            dq.push_front(*it);
        }

        return vector<int>(dq.begin(), dq.end());
    }
};